### PR TITLE
Adding update button

### PR DESCRIPTION
**Pull request by timmillwood**

### CODE REVIEW

File: src/Form/UpdateForm.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+   */
+  protected function getUpstream() {
+    $workspace = \Drupal::service('workspace.manager')->getActiveWorkspace();
+    if (isset($workspace->upstream)) {
+      return $workspace->upstream->entity;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
```
Comment (jeqq) [inline src/Form/UpdateForm.php]: Let's keep this consistent with the Drupal core and use `{@inheritdoc}` instead (lowercase).

File: src/Form/UpdateForm.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      $response = \Drupal::service('workspace.replicator_manager')->update(
+        $this->getUpstream(),
+        $this->getActive()
+      );
+
+      if (($response instanceof ReplicationLogInterface) && $response->get('ok')) {
+        drupal_set_message('Successful update.');
+      }
+      else {
+        drupal_set_message('Error', 'error');
```
Comment (jeqq) [inline src/Form/UpdateForm.php]: Maybe we should consider a better error message here?

File: src/Form/UpdateForm.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  /**
+   * {@inheritDoc}
+   */
+  public function getDescription() {
+    if (!$this->getUpstream()) {
+      return $this->t('%workspace has no upstream set.', ['%workspace' => $this->getActive()->label()]);
+    }
+    return $this->t('Do you want to pull changes from %upstream to %workspace?', ['%upstream' => $this->getUpstream()->label(), '%workspace' => $this->getActive()->label()]);
+  }
+}
```
Comment (jeqq) [inline src/Form/UpdateForm.php]: Missing new line at EOF.

File: src/Form/UpdateForm.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    try {
+      $response = \Drupal::service('workspace.replicator_manager')->update(
+        $this->getUpstream(),
+        $this->getActive()
+      );
+
+      if (($response instanceof ReplicationLogInterface) && $response->get('ok')) {
+        drupal_set_message('Successful update.');
```
Comment (dickolsson) [inline src/Form/UpdateForm.php:92]: Let's provide a more useful message. Something like `The current workspace [name] was updated with content from upstream workspace [name].`

**Pull request merged**